Sources/SwiftSyntaxBuilder/ConvenienceInitializers.swift

```diff
@@ -141,12 +141,11 @@ extension FloatLiteralExprSyntax: ExpressibleByFloatLiteral {
 // MARK: - FunctionCallExpr
 
 extension FunctionCallExpr {
-  // Need an overload that's explicitly `ExprSyntax` for code literals to work.
   /// A convenience initializer that allows passing in arguments using a result builder
   /// instead of having to wrap them in a `TupleExprElementList`.
   /// The presence of the parenthesis will be inferred based on the presence of arguments and the trailing closure.
-  public init(
-    callee: ExprSyntax,
+  public init<C: ExprSyntaxProtocol>(
+    callee: C,
```

Comment:
Note: this means that we can't do `FunctionCallExpr(callee: "foo"...)` any more. Do we want to leave in the extra `ExprSyntax` overload so that doesn't have to be `FunctionCallExprSyntax(callee: ExprSyntax("Foo")...)`?

If so... maybe we should have those initializers in general, ie. the generic init + overload taking base types as well. Thoughts @ahoppen / @rintaro?

Reply:
I've replaced all our calls for now. Let me know what you think (all wrapped in `ExprSyntax` now).